fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -601,7 +601,8 @@ public void execute(TUniqueId queryId) throws Exception {
                     if (e instanceof NereidsException && !context.getSessionVariable().enableFallbackToOriginalPlanner
                             && !forceFallback) {
                         LOG.warn("Analyze failed. {}", context.getQueryIdentifier(), e);
-                        throw ((NereidsException) e).getException();
+                        context.getState().setError(e.getMessage());
```

Comment:
If the code like this:

statementexecutor.execute();
code1;
code2;

if you not throw exception,then code1 and code2 will run. BUT it will not run in the past.
Any problems?

Reply:
no, this is what we expected. U could see we return directly after setError. in the function caller it will process correclty after we do setError.